Introducing @aws/nx-plugin:ts#sync causes Nx to ignore the default @nx/js:typescript-sync

### Current Behavior

Nx treats an explicit syncGenerators list as an override so `nx build` no longer triggers `@nx/js:typescript-sync`

### Expected Behavior

`nx build` should trigger `@aws/nx-plugin:ts#sync` and the default `@nx/js:typescript-sync`

### Steps to Reproduce

1. Scaffold a new project 
```
npx create-nx-workspace@22.1.3 dungeon-adventure --pm=pnpm  --preset=@aws/nx-plugin --iacProvider=CDK --ci=skip --aiAgents
```
 
2. Create a new trpc#api, this will generate the api and `common/constructs`

```
pnpm exec nx generate @aws/nx-plugin:ts#trpc-api --name=MyApi --no-interactive
```

3. Build the project, this will fail because `@nx/js:typescript-sync` needs to run to update `packages/common/constructs/tsconfig.lib.json > references` and include the path to the api project.

```
pnpm nx run-many --target build --all
```

4. Update `nx.json` to either remove `targetDefaults.compile.syncGenerators` or update it to `["@aws/nx-plugin:ts#sync", "@nx/js:typescript-sync"]` and the build will succeed.

### Nx Report

```shell
error TS6059: File '/home/alex/Projects/my-proj/packages/api/src/schema/index.ts' is not under 'rootDir' '/home/alex/Projects/my-proj/packages/common/constructs'. 'rootDir' is expected to contain all source files.
```

### Failure Logs

```shell

```

### Additional Information

_No response_

--- a/packages/nx-plugin/src/preset/generator.spec.ts
+++ b/packages/nx-plugin/src/preset/generator.spec.ts
@@ -8,6 +8,8 @@ import { createTreeUsingTsSolutionSetup, snapshotTreeDir } from '../utils/test';
 import { vi, describe, it, expect, beforeEach } from 'vitest';
 import { SYNC_GENERATOR_NAME as TS_SYNC_GENERATOR_NAME } from '../ts/sync/generator';
 
+const NX_TYPESCRIPT_SYNC_GENERATOR = '@nx/js:typescript-sync';
+
 // Mock execSync to control git command behavior in tests
 vi.mock('child_process', async (importOriginal) => {
   const actual = await importOriginal<typeof import('child_process')>();
@@ -54,12 +56,14 @@ describe('preset generator', () => {
     expect((await readAwsNxPluginConfig(tree)).iac.provider).toBe('CDK');
   });
 
-  it('should register the TypeScript sync generator for compile targets', async () => {
+  it('should register the TypeScript sync generators for compile targets', async () => {
     await presetGenerator(tree, { addTsPlugin: false, iacProvider: 'CDK' });
 
-    expect(readNxJson(tree).targetDefaults?.compile?.syncGenerators).toContain(
-      TS_SYNC_GENERATOR_NAME,
-    );
+    const syncGenerators =
+      readNxJson(tree).targetDefaults?.compile?.syncGenerators;
+
+    expect(syncGenerators).toContain(TS_SYNC_GENERATOR_NAME);
+    expect(syncGenerators).toContain(NX_TYPESCRIPT_SYNC_GENERATOR);
   });
 });
 

--- a/packages/nx-plugin/src/preset/generator.ts
+++ b/packages/nx-plugin/src/preset/generator.ts
@@ -33,6 +33,7 @@ import {
 import { SYNC_GENERATOR_NAME as TS_SYNC_GENERATOR_NAME } from '../ts/sync/generator';
 
 const WORKSPACES = ['packages/*'];
+const NX_TYPESCRIPT_SYNC_GENERATOR = '@nx/js:typescript-sync';
 
 export const PRESET_GENERATOR_INFO: NxGeneratorInfo =
   getGeneratorInfo(__filename);
@@ -154,8 +155,12 @@ export const presetGenerator = async (
         ...nxJson.targetDefaults?.compile,
         syncGenerators: [
           ...(nxJson.targetDefaults?.compile?.syncGenerators ?? []).filter(
-            (g) => g !== TS_SYNC_GENERATOR_NAME,
+            (g) =>
+              ![TS_SYNC_GENERATOR_NAME, NX_TYPESCRIPT_SYNC_GENERATOR].includes(
+                g,
+              ),
           ),
+          NX_TYPESCRIPT_SYNC_GENERATOR,
           TS_SYNC_GENERATOR_NAME,
         ],
       },

